hw/xbox/xid-sdl.c

```diff
@@ -511,18 +582,11 @@ static void usb_xbox_gamepad_realize(USBDevice *dev, Error **errp)
     SDL_JoystickEventState(SDL_ENABLE);
 #endif
 
-#ifdef FORCE_FEEDBACK
-    s->sdl_haptic = SDL_HapticOpenFromJoystick(s->sdl_gamepad);
-    if (s->sdl_haptic == NULL) {
-        fprintf(stderr, "Joystick doesn't support haptic\n");
+    if (SDL_InitSubSystem(SDL_INIT_HAPTIC)) {
+        fprintf(stderr, "SDL failed to initialize haptic feedback subsystem\n");
     } else {
-        if ((SDL_HapticQuery(s->sdl_haptic) & SDL_HAPTIC_LEFTRIGHT) == 0) {
-            fprintf(stderr, "Joystick doesn't support necessary haptic effect\n");
-            SDL_HapticClose(s->sdl_haptic);
-            s->sdl_haptic = NULL;
-        }
+        haptic_init(s);
```

Comment:
nit: I think we usually do: `if (noterror) { do stuff } else { handle-error }`.

I might be wrong though, but it would certainly make this bit of code easier to follow.